exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/AbstractParquetScanBatchCreator.java

```diff
@@ -68,76 +84,144 @@ protected ScanBatch getBatch(ExecutorFragmentContext context, AbstractParquetRow
-        }
+          Stopwatch timer = logger.isTraceEnabled() ? Stopwatch.createUnstarted() : null;
+          DrillFileSystem fs = fsManager.get(rowGroupScan.getFsConf(rowGroup), rowGroup.getPath());
+          if (!footers.containsKey(rowGroup.getPath())) {
+            if (timer != null) {
+              timer.start();
+            }
+
+            ParquetMetadata footer = readFooter(fs.getConf(), rowGroup.getPath(), readerConfig);
+            if (timer != null) {
+              long timeToRead = timer.elapsed(TimeUnit.MICROSECONDS);
+              logger.trace("ParquetTrace,Read Footer,{},{},{},{},{},{},{}", "", rowGroup.getPath(), "", 0, 0, 0, timeToRead);
+            }
+            footers.put(rowGroup.getPath(), footer);
+          }
+          ParquetMetadata footer = footers.get(rowGroup.getPath());
+
+          //
+          //   If a filter is given (and it is not just "TRUE") - then use it to perform run-time pruning
+          //
+          if ( doRuntimePruning  ) { // skip when no filter or filter is TRUE
+
+            int rowGroupIndex = rowGroup.getRowGroupIndex();
+            long footerRowCount = footer.getBlocks().get(rowGroupIndex).getRowCount();
+
+            if ( timer != null ) {  // restart the timer, if tracing
+              timer.reset();
+              timer.start();
+            }
+
+            // When starting a new file, or at the first time - Initialize the path specific metadata
+            if ( ! rowGroup.getPath().equals(prevRowGroupPath) ) {
+              // Create a table metadata (V4)
+              tableMetadataV4 = new Metadata_V4.ParquetTableMetadata_v4();
+
+              // The file status for this file
+              FileStatus fileStatus = fs.getFileStatus(rowGroup.getPath());
+
+              // The file metadata (only for the columns used in the filter)
+              fileMetadataV4 = Metadata.getParquetFileMetadata_v4(tableMetadataV4, footer, fileStatus, fs, false, true, columnsInExpr, readerConfig);
+
+              prevRowGroupPath = rowGroup.getPath(); // for next time
+            }
+
+            MetadataBase.RowGroupMetadata rowGroupMetadata = fileMetadataV4.getFileMetadata().getRowGroups().get(rowGroup.getRowGroupIndex());
+
+            Map<SchemaPath, ColumnStatistics> columnsStatistics = ParquetTableMetadataUtils.getRowGroupColumnStatistics(tableMetadataV4, rowGroupMetadata);
 
-      } catch (IOException e) {
-        throw new ExecutionSetupException(e);
+            //
+            // Perform the Run-Time Pruning - i.e. Skip this rowgroup if the match fails
+            //
+            RowsMatch match = FilterEvaluatorUtils.matches(filterPredicate, columnsStatistics, footerRowCount);
+            if (timer != null) { // if tracing
+              long timeToRead = timer.elapsed(TimeUnit.MICROSECONDS);
+              logger.trace("Run-time pruning: {} row-group {} (RG index: {} row count: {}), took {} usec", match == RowsMatch.NONE ? "Excluded" : "Included", rowGroup.getPath(),
```

Comment:
Ok to log this at the TRACE level but could you add an INFO level log ***outside*** the for loop indicating how many row groups were evaluated and how many were pruned out.  Even though the operator profile has the metrics,  the log would be quite useful.

Reply:
Please also add a keyword such as PARQUET_RUNTIME_PRUNING so that the QA test framework can find this log message and verify it.

Reply:
Done, in a followup commit

Reply:
It does not look like any INFO level logging was added..I still see only TRACE and DEBUG levels. 
Also, @rhou1's suggestion for annotating the info message with a prefix seems reasonable for testing purposes.